webapp/src/utils/new-format-objects.js

```diff
@@ -105,37 +57,144 @@ export const newDataFormatPercentByElection = (
+        eosTotal += electionsList[pos].amount
+      } else {
+        const election = {
+          election: `Election ${electionsList[pos].election + 1}`,
+          EOS_TOTAL: Number(eosTotal + electionsList[pos].amount),
+          USD_TOTAL: Number(usdTotal + electionsList[pos].usd_total),
+          EOS_CATEGORIZED: Number(eosTotal),
+          USD_CATEGORIZED: Number(usdTotal),
+          EOS_UNCATEGORIZED: Number(electionsList[pos].amount),
+          USD_UNCATEGORIZED: Number(electionsList[pos].usd_total)
+        }
+        elections.push(election)
+        eosTotal = 0
+        usdTotal = 0
+        electionNum++
+      }
+    } else {
+      const election = {
+        election: `Election ${electionsList[pos].electionNum + 1}`,
+        EOS_TOTAL: Number(eosTotal + electionsList[pos].amount),
+        USD_TOTAL: Number(usdTotal + electionsList[pos].usd_total),
+        EOS_CATEGORIZED: Number(eosTotal),
+        USD_CATEGORIZED: Number(usdTotal),
+        EOS_UNCATEGORIZED: Number(electionsList[pos].amount),
+        USD_UNCATEGORIZED: Number(electionsList[pos].usd_total)
+      }
+      elections.push(election)
+      eosTotal = 0
+      usdTotal = 0
+      electionNum++
+      pos--
+    }
+  }
+  return elections
+}
+
+export const newDataFormatByAllDelegates = transactionsList =>
   transactionsList.map(data => ({
     name: data.delegate_payer,
-    EOS: Number(data.amount),
-    USD: Number(data.usd_total),
+    EOS_CATEGORIZED: Number(data.amount),
+    USD_CATEGORIZED: Number(data.usd_total),
     color: generateColor()
   }))
 
-export const newDataFormatByElectionAndDelegate = (
-  transactionsList,
-  isIncome
-) =>
-  isIncome
-    ? transactionsList.map(data => ({
-        name: data.recipient,
-        EOS_CLAIMED: Number(data.eos_claimed),
-        USD_CLAIMED: Number(data.usd_claimed),
-        EOS_UNCLAIMED: Number(data.eos_unclaimed),
-        USD_UNCLAIMED: Number(data.usd_unclaimed)
-      }))
-    : transactionsList.map(data => ({
-        name: data.delegate_payer,
-        EOS_CATEGORIZED: Number(data.eos_categorized),
-        USD_CATEGORIZED: Number(data.usd_categorized),
-        EOS_UNCATEGORIZED: Number(data.eos_uncategorized),
-        USD_UNCATEGORIZED: Number(data.usd_uncategorized)
-      }))
+export const newDataFormatByElectionAndDelegate = transactionsList =>
+  transactionsList.map(data => ({
+    name: data.delegate_payer,
+    EOS_CATEGORIZED: Number(data.eos_categorized),
+    USD_CATEGORIZED: Number(data.usd_categorized),
+    EOS_UNCATEGORIZED: Number(data.eos_uncategorized),
+    USD_UNCATEGORIZED: Number(data.usd_uncategorized),
+    color: generateColor()
+  }))
+
+export const newDataFormatTotalByCategory = totalByCategory =>
+  totalByCategory.map(data => ({
+    name: data.category ? data.category : `Election ${data.election + 1}`,
+    EOS_CATEGORIZED: Number(data.amount),
+    USD_CATEGORIZED: Number(data.usd_total),
+    color: generateColor()
+  }))
+
+// Delegate
```

Comment:
We need this comment

Reply:
Is for know by view although can add the name view and delete the comment